compute/local/backend.go

```diff
@@ -46,9 +67,18 @@ func (b *Backend) Submit(task *tes.Task) error {
 		return err
 	}
 
-	go func() {
-		w.Run(ctx)
+	go func() { 
+		// WaitGroup to block closing the worker until the last event is written from events/executor.go:StreamLogTail()
+		var wg sync.WaitGroup
```

Comment:
Seems like it would make more sense to create the `WaitGroup` in worker.go and just ensure that `Run` blocks until all logs are written.

Reply:
Actually, the simplest place to get the same behavior would be adding the wait group to step.go and then adding the wait call somewhere around [here](https://github.com/ohsu-comp-bio/funnel/blob/master/worker/step.go#L54-L68).

Reply:
Good call — moved the WaitGroup closer to the output event writer routine that was causing the panic after the session closed* ([executor.go](https://github.com/ohsu-comp-bio/funnel/blob/fix/lint/events/executor.go#L238-L251))

Tried adding the wait call to [step.go](https://github.com/ohsu-comp-bio/funnel/blob/master/worker/step.go#L54-L68) but encountering the same panic result — does `<-done` wait for the final event writer to complete?

---

*Session close panic that prompted adding the WaitGroup: [Actions Run](https://github.com/ohsu-comp-bio/funnel/actions/runs/5514073875/job/14928495674#step:4:438)

Reply:
The <-done is only going to wait for the command to exit. You need to wait group call after that for it to block.